File: 3.2/mobile/lab2/lab2/project2/app/src/main/java/com/example/project2/MainActivity.java
package com.example.project2;

import android.os.Bundle;
import android.view.View;
import android.widget.*;
import androidx.appcompat.app.AppCompatActivity;
import java.util.Random;

public class MainActivity extends AppCompatActivity {
    private TextView tvInfo;
    private EditText etInput;
    private Button bControl;
    private int secretNumber;
    private boolean gameFinished;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        tvInfo = findViewById(R.id.tvInfo);
        etInput = findViewById(R.id.etInput);
        bControl = findViewById(R.id.bControl);

        startNewGame();
    }

    private void startNewGame() {
        secretNumber = new Random().nextInt(200) + 1;
        gameFinished = false;
        tvInfo.setText(getString(R.string.try_to_guess));
        bControl.setText(getString(R.string.play_more));
        etInput.setText("");
    }

    public void onClick(View v) {
        if (gameFinished) {
            startNewGame();
            return;
        }

        String inputText = etInput.getText().toString().trim();

        // Проверка: если поле ввода пустое
        if (inputText.isEmpty()) {
            Toast.makeText(this, getString(R.string.error_no_input), Toast.LENGTH_SHORT).show();
            return;
        }

        int userGuess;
        try {
            userGuess = Integer.parseInt(inputText);
        } catch (NumberFormatException e) {
            // Если введено не число
            Toast.makeText(this, getString(R.string.error_not_a_number), Toast.LENGTH_SHORT).show();
            return;
        }

        // Проверка диапазона (число должно быть от 0 до 200)
        if (userGuess < 0 || userGuess > 200) {
            Toast.makeText(this, getString(R.string.error_out_of_range), Toast.LENGTH_SHORT).show();
            return;
        }

        // Логика сравнения введенного числа с загаданным
        if (userGuess < secretNumber) {
            tvInfo.setText(getString(R.string.ahead));
        } else if (userGuess > secretNumber) {
            tvInfo.setText(getString(R.string.behind));
        } else {
            tvInfo.setText(getString(R.string.hit));
            gameFinished = true;
            bControl.setText(getString(R.string.play_more));
        }

        etInput.setText("");
    }
}

File: 3.2/mobile/lab4/lab4/project3/app/src/main/java/com/example/project3/MainActivity.kt
package com.example.project3

import android.gesture.Gesture
import android.gesture.GestureLibraries
import android.gesture.GestureLibrary
import android.gesture.GestureOverlayView
import android.os.Bundle
import android.view.View
import android.widget.Button
import android.widget.EditText
import android.widget.TextView
import androidx.appcompat.app.AppCompatActivity

class MainActivity : AppCompatActivity(), GestureOverlayView.OnGesturePerformedListener {

    private lateinit var tvInfo: TextView
    private lateinit var etInput: EditText
    private lateinit var bControl: Button
    private var targetNumber: Int = 0
    private var gameFinished: Boolean = false
    private lateinit var gestureLibrary: GestureLibrary

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        tvInfo = findViewById(R.id.textView1)
        etInput = findViewById(R.id.input_value)
        bControl = findViewById(R.id.submit)

        targetNumber = (1..100).random()
        gameFinished = false

        gestureLibrary = GestureLibraries.fromRawResource(this, R.raw.gestures)
        if (!gestureLibrary.load()) {
            finish()
        }

        val gestureOverlayView = findViewById<GestureOverlayView>(R.id.gestureOverlay)
        gestureOverlayView.addOnGesturePerformedListener(this)
    }
    override fun onGesturePerformed(overlay: GestureOverlayView, gesture: Gesture) {
        val predictions = gestureLibrary.recognize(gesture)
        if (predictions.isNotEmpty()) {
            val prediction = predictions[0]
            if (prediction.score > 1.0) {
                when (prediction.name) {
                    "0" -> processDigit(0)
                    "1" -> processDigit(1)
                    "2" -> processDigit(2)
                    "3" -> processDigit(3)
                    "4" -> processDigit(4)
                    "5" -> processDigit(5)
                    "6" -> processDigit(6)
                    "7" -> processDigit(7)
                    "8" -> processDigit(8)
                    "9" -> processDigit(9)
                    "stop" -> finalizeInput()
                    else -> tvInfo.text = resources.getString(R.string.error)
                }
            }
        }
    }

    private fun processDigit(digit: Int) {
        etInput.append(digit.toString())
    }

    private fun finalizeInput() {
        onClick(bControl)
    }


    fun onClick(v: View) {
        try {
            if (!gameFinished) {
                val inputText = etInput.text.toString()
                if (inputText.isEmpty()) {
                    tvInfo.text = resources.getString(R.string.error)
                }

                val inputNumber = inputText.toInt()
                when {
                    inputNumber < 1 || inputNumber > 100 -> {
                        tvInfo.text = resources.getString(R.string.error)
                    }
                    inputNumber < targetNumber -> {
                        tvInfo.text = resources.getString(R.string.behind)
                    }
                    inputNumber > targetNumber -> {
                        tvInfo.text = resources.getString(R.string.ahead)
                    }
                    else -> {
                        tvInfo.text = resources.getString(R.string.hit)
                        bControl.text = resources.getString(R.string.play_more)
                        gameFinished = true
                    }
                }
            } else {
                targetNumber = (1..100).random()
                tvInfo.text = resources.getString(R.string.try_to_guess)
                bControl.text = resources.getString(R.string.input_value)
                gameFinished = false
            }
        } catch (e: NumberFormatException) {
            tvInfo.text = resources.getString(R.string.error)
        }

        etInput.setText("")
    }
}
